src/to_substrait.cpp

```diff
@@ -1391,6 +1430,90 @@ substrait::Rel *DuckDBToSubstrait::TransformIntersect(LogicalOperator &dop) {
 	return rel;
 }
 
+substrait::Rel *DuckDBToSubstrait::TransformCreateTable(LogicalOperator &dop) {
+	auto rel = new substrait::Rel();
+	auto &create_table = dop.Cast<LogicalCreateTable>();
+	auto &create_info = create_table.info.get()->Base();
+
+	auto schema = new substrait::NamedStruct();
+	auto type_info = new substrait::Type_Struct();
+	type_info->set_nullability(substrait::Type_Nullability_NULLABILITY_REQUIRED);
+	for (auto &name : create_info.columns.GetColumnNames()) {
+		schema->add_names(name);
+	}
+	for (auto &col_type : create_info.columns.GetColumnTypes()) {
+		auto s_type = DuckToSubstraitType(col_type, nullptr, false);
+		*type_info->add_types() = s_type;
+	}
+	schema->set_allocated_struct_(type_info);
+
+	if (create_table.children.size() == 0) {
+		// This is create table with schema
+		auto ddl = rel->mutable_ddl();
+		ddl->set_op(substrait::DdlRel::DdlOp::DdlRel_DdlOp_DDL_OP_CREATE);
+		ddl->set_object(substrait::DdlRel::DDL_OBJECT_TABLE);
+		ddl->set_allocated_table_schema(schema);
+
+		auto named_object = ddl->mutable_named_object();
+		named_object->add_names(create_info.schema);
+		named_object->add_names(create_info.table);
+		return rel;
+	}
+
+	// This is create table as select
+	substrait::Rel *input = nullptr;
+	switch (create_table.children[0]->type)
+	{
+	case LogicalOperatorType::LOGICAL_PROJECTION:
+		input = TransformProjection(*create_table.children[0]);
+		break;
+	default:
+		throw NotImplementedException("Create table with more than one child");
```

Comment:
can we turn this into a message that would be clearer to an end user? What would generate this situation?